test/e2e/proxy_test.go

```diff
@@ -68,7 +70,493 @@ func skipIfProxyDisabled(t *testing.T) {
+		testNodePortLocalFromLocal(t, data, localUrlCp, localUrlWk, localUrlLo, testPodHostnameCp, testPodHostnameWk)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Local Client:Pod", func(t *testing.T) {
+		testNodePortLocalFromPod(t, data, localUrlCp, localUrlWk, clientPodCp, clientPodWk, podClientCpIP, podClientWkIP, testPodHostnameCp, testPodHostnameWk)
+	})
+}
+
+func createTestAgnhostPods(t *testing.T, data *TestData, agnhostCp, agnhostWk string, hostNetwork bool) {
+	// Create test agnhost pod on each node.
+	args := []string{"netexec", "--http-port=8080"}
+	ports := []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: 8080,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+	if agnhostCp != "" {
+		require.NoError(t, data.createPodOnNode(agnhostCp, testNamespace, nodeName(0), agnhostImage, []string{}, args, nil, ports, hostNetwork, nil))
+		_, err := data.podWaitForIPs(defaultTimeout, agnhostCp, testNamespace)
+		require.NoError(t, err)
+		require.NoError(t, data.podWaitForRunning(defaultTimeout, agnhostCp, testNamespace))
+	}
+	if agnhostWk != "" {
+		require.NoError(t, data.createPodOnNode(agnhostWk, testNamespace, nodeName(1), agnhostImage, []string{}, args, nil, ports, hostNetwork, nil))
+		_, err := data.podWaitForIPs(defaultTimeout, agnhostWk, testNamespace)
+		require.NoError(t, err)
+		require.NoError(t, data.podWaitForRunning(defaultTimeout, agnhostWk, testNamespace))
+	}
+}
+
+func createTestClientPods(t *testing.T, data *TestData, clientCp, clientWk string) (string, string, string, string) {
+	// Create a busybox Pod on each node which is used as test client.
+	require.NoError(t, data.createBusyboxPodOnNode(clientCp, testNamespace, nodeName(0)))
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, clientCp, testNamespace))
+	busyboxCpPod, err := data.podWaitFor(defaultTimeout, clientCp, testNamespace, func(pod *corev1.Pod) (bool, error) {
+		return pod.Status.Phase == corev1.PodRunning, nil
+	})
+	require.NoError(t, err)
+	require.NotNil(t, busyboxCpPod.Status)
+	require.NoError(t, data.createBusyboxPodOnNode(clientWk, testNamespace, nodeName(1)))
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, clientWk, testNamespace))
+	busyboxWkPod, err := data.podWaitFor(defaultTimeout, clientWk, testNamespace, func(pod *corev1.Pod) (bool, error) {
+		return pod.Status.Phase == corev1.PodRunning, nil
+	})
+	require.NoError(t, err)
+	require.NotNil(t, busyboxWkPod.Status)
+
+	cpPodIPv4 := busyboxCpPod.Status.PodIP
+	wkPodIPv4 := busyboxWkPod.Status.PodIP
+
+	var cpPodIPv6, wkPodIPv6 string
+	for _, ip := range busyboxCpPod.Status.PodIPs {
+		if utilnet.IsIPv6(net.ParseIP(ip.IP)) {
+			cpPodIPv6 = ip.IP
+			break
+		}
+	}
+	for _, ip := range busyboxWkPod.Status.PodIPs {
+		if utilnet.IsIPv6(net.ParseIP(ip.IP)) {
+			wkPodIPv6 = ip.IP
+			break
+		}
+	}
+
+	return cpPodIPv4, wkPodIPv4, cpPodIPv6, wkPodIPv6
+}
+
+func deleteTestAgnhostPods(t *testing.T, data *TestData, agnhostCp, agnhostWk string) {
```

Comment:
ditto, maybe the problem exists for all functions in this file. It doesn't make sense to create two pod versions of create/delete functions just for this test, which looks rather redundant.

Reply:
Updated.